drivers/overlay/overlay.go

```diff
@@ -1657,7 +1664,15 @@ func (d *Driver) ApplyDiffFromStagingDirectory(id, parent, stagingDirectory stri
 	if err := os.RemoveAll(diff); err != nil && !os.IsNotExist(err) {
 		return err
 	}
-	return os.Rename(stagingDirectory, diff)
+
+	err = os.Rename(stagingDirectory, diff)
+	if err != nil && errors.Is(err, unix.EXDEV) {
+		// If we got EXDEV, copy the project quota from the staging directory to the target path and try again.
```

Comment:
I think a comment like “EXDEV might mean that we failed because $reason, so $fix_that_reason” would be much more useful than just a comment trivially re-stating what the code below does.